src/client/pages/Offer/ProductSelector/ProductSlector.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useState, useRef, useMemo } from 'react'
+import styled from '@emotion/styled'
+import { useTextKeys } from 'utils/textKeys'
+import { MEDIA_QUERIES } from 'utils/mediaQueries'
+import { MainProductCard } from './MainProductCard'
+import { AdditionalProductCard } from './AdditionalProductCard'
+
+type Product = {
+  id: string
+  name: string
+  description: string
+  price: string
+  image: string
+}
+
+type ProductSelectorProps = {
+  className?: string
+  mainProducts: Array<Product>
+  additionalProducts: Array<Product>
+}
+
+const getInitialState = ({
+  mainProducts,
+  additionalProducts,
+}: {
+  mainProducts: Array<Product>
+  additionalProducts: Array<Product>
+}): {
+  mainProducts: Record<string, boolean>
+  additionalProducts: Record<string, boolean>
+} => {
+  const getProductIdByValueMap = (products: Array<Product>) =>
+    products.reduce(
+      (result, product) => ({
+        ...result,
+        [product.id]: false,
+      }),
+      {},
+    )
+
+  return {
+    mainProducts: getProductIdByValueMap(mainProducts),
+    additionalProducts: getProductIdByValueMap(additionalProducts),
+  }
+}
+
+export const ProductSelector = ({
+  className,
+  mainProducts,
+  additionalProducts,
+}: ProductSelectorProps) => {
+  const textKeys = useTextKeys()
+  const inputRef = useRef<HTMLInputElement | null>(null)
+  const [selectedProductsByCategory, setSelectedProductsByCategory] = useState(
+    getInitialState({ mainProducts, additionalProducts }),
+  )
+
+  const selectedMainProducts = useMemo(() => {
+    return Object.keys(selectedProductsByCategory.mainProducts).filter(
+      (productId) => selectedProductsByCategory.mainProducts[productId],
+    )
+  }, [selectedProductsByCategory])
+
+  return (
+    <Wrapper className={className}>
+      <section>
+        <h2>Main Coverage</h2>
+        <MainCoverageCardGrid>
+          {mainProducts.map(({ id, name, price, description, image }) => {
+            const isTheOnlySelectedMainProduct =
+              selectedMainProducts.length === 1 &&
+              selectedMainProducts[0] === id
+
+            return (
+              <MainProductCard
+                key={id}
+                checkboxRef={
+                  isTheOnlySelectedMainProduct ? inputRef : undefined
+                }
```

Comment:
What are we trying to do here?

Reply:
In order to show an error message using the native Web api  (`element.[setCustomValidity()/reportValidity()]`) I need to hold a reference to checkbox that belongs to only currently selected main product. That's because in case the user clicks on that card with the intention of unselecting it I can use that reference and call the API to show the error message.

Reply:
Right, I see the reason now. Seems a little non-standard to assign the ref conditionally, though.

Another approach would be to assign refs to each checkbox and later call "setCustomValidity" on the relevant one. But okey since this is the web-onboarding repo hehe

Reply:
I've tried a couple of different approaches but they appear more confusing 😢 
So I think I'm gonna leave it like that and iterate over it later.